File: 3770_Largest_Prime_from_Consecutive_Prime_Sum/solution.c
int largestPrime(int n) {
    if (n < 2) return 0;
    
    bool* is_prime = (bool*)malloc((n + 1) * sizeof(bool));
    for (int i = 0; i <= n; i++) is_prime[i] = true;
    is_prime[0] = is_prime[1] = false;

    for (int i = 2; i * i <= n; i++)
        if (is_prime[i]) for (int j = i * i; j <= n; j += i) is_prime[j] = false;
    
    int* primes = (int*)malloc((n + 1) * sizeof(int));
    int count = 0;
    for (int i = 2; i <= n; i++) if (is_prime[i]) primes[count++] = i;

    int result = 0, sum = 0;
    for (int i = 0; i < count; i++) {
        sum += primes[i];
        if (sum > n) break;
        if (is_prime[sum]) result = sum;
    }

    free(is_prime);
    free(primes);
    return result;
}

File: 3770_Largest_Prime_from_Consecutive_Prime_Sum/solution.py
class Solution:
    def largestPrime(self, n: int) -> int:
        if n < 2: return 0

        is_prime = [True] * (n + 1)
        is_prime[0] = is_prime[1] = False

        for i in range(2, int(n**0.5) + 1):
            if is_prime[i]:
                for j in range(i * i, n + 1, i):
                    is_prime[j] = False

        primes = [i for i in range(2, n + 1) if is_prime[i]]
        result, sum_val = 0, 0

        for prime in primes:
            sum_val += prime

            if sum_val > n: break

            if is_prime[sum_val]: result = sum_val

        return result